clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -325,4 +333,21 @@ void cleanup() {
             log.debug("Closed the consumer network thread");
         }
     }
+
+    /**
+     * If there is a metadata error, completed all uncompleted events with the metadata error.
```

Comment:
```suggestion
     * If there is a metadata error, complete all uncompleted events that require subscription metadata
```

something like it, to clarify that it's only for the ones requiring metadata